tests/System.Text.JsonLab.Tests/JsonReaderTests.cs

```diff
@@ -308,22 +308,141 @@ public static void TestJsonReaderUtf8SpecialString(string jsonString, string exp
             Assert.Equal(expectedStr, actualStr);
         }
 
+
+        // TODO: Need a mechanism to pass state appropriately
+        [Theory]
+        [InlineData("\"hello\"", 1, 0)] // "\""
+        [InlineData("12345", 3, 0)]   // "123"
+        [InlineData("null", 3, 0)]   // "nul"
+        [InlineData("true", 3, 0)]   // "tru"
+        [InlineData("false", 4, 0)]  // "fals"
+        //[InlineData("{\"age\":30}", 7, 7)] // "{\"age\":"
+        //[InlineData("{\"name\":\"Ahson\"", 9, 8)]  // "{\"name\":\"Ahso"
+        [InlineData("-123456789", 1, 0)] // "-"
+        [InlineData("0.5", 2, 0)]    // "0."
+        [InlineData("10.5e+3", 5, 0)] // "10.5e"
+        [InlineData("10.5e-1", 6, 0)]    // "10.5e-"
+        //[InlineData("{\"ints\":[1, 2, 3, 4, 5]}", 21, 21)]    // "{\"ints\":[1, 2, 3, 4, 5"
+        //[InlineData("{\"strings\":[\"abc\", \"def\"], \"ints\":[1, 2, 3, 4, 5]}", 24, 24)]  // "{\"strings\":[\"abc\", \"def\""
+        //[InlineData("{\"age\":30, \"name\":\"test}:[]\", \"another string\" : \"tests\"}", 19, 18)]   // "{\"age\":30, \"name\":\"test}"
+        public static void SplitJsonAsync(string jsonString, int splitLocation, int consumed)
+        {
+            var buffer = new char[jsonString.Length];
+            var reader = new StringReader(jsonString);
+
+            reader.ReadAsync(buffer, 0, splitLocation);
```

Comment:
You don't happen to need to await here (as the call returns synchronously, but this is very fragile. 

Having said that, I would not do actual async calls in all unit test like this. I think it would be enough to simulate continuing with the reader by first parsing partial data and then parsing the rest with a new reader.